// Copyright 2024 MozoLM Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Message encapsulating information about the model storage.
//
// For most of the models one may simply require the actual model file. In
// addition, some models may store the vocabulary separately, while other
// models may require additional data.

syntax = "proto3";

package mozolm;

import "mozolm/models/ngram_word_fst_options.proto";
import "mozolm/models/ppm_as_fst_options.proto";

option java_outer_classname = "ModelStorageProto";

// Next available ID: 5
message ModelStorage {
  // Actual model file. For example, this can be a text file in ARPA format or
  // a binary file in some other format.
  string model_file = 1;

  // Vocabulary file. This field will be ignored if the model already contains
  // the vocabulary.
  string vocabulary_file = 2;

  // Options for PPM model.
  PpmAsFstOptions ppm_options = 3;

  // Options for NGramWordFst model.
  NGramWordFstOptions ngram_word_fst_options = 4;
}
